plugins/module_utils/rds.py

```diff
@@ -492,6 +502,43 @@ def wait_for_status(client, module: AnsibleAWSModule, identifier: str, method_na
         wait_for_cluster_snapshot_status(client, module, identifier, waiter_name)
 
 
+def get_snapshot(
+    client, module: AnsibleAWSModule, snapshot_identifier: str, snapshot_type: str, convert_tags: bool = True
+) -> Dict[str, Any]:
+    """
+    Returns instance or cluster snapshot attributes given the snapshot identifier.
+
+        Parameters:
+            client: boto3 rds client
+            module: AnsibleAWSModule
+            snapshot_identifier (str): Unique snapshot identifier
+            snapshot_type (str): Which type of snapshot to get, one of: cluster, instance
+            convert_tags (bool): Whether to convert the snapshot tags from boto3 list of dicts to Ansible dict; defaults to True
+
+        Returns:
+            snapshot (dict): Snapshot attributes. If snapshot with provided id is not found, returns an empty dict
+
+        Raises:
+            Failes the module if an exception is raised while retrieving the snapshot attributes.
+    """
+    snapshot = {}
+
+    try:
+        if snapshot_type == "cluster":
+            snapshots = describe_db_cluster_snapshots(client, DBClusterSnapshotIdentifier=snapshot_identifier)
+        elif snapshot_type == "instance":
+            snapshots = describe_db_snapshots(client, DBSnapshotIdentifier=snapshot_identifier)
+        if snapshots:
```

Comment:
`snapshots` should be initialized somewhere, otherwise, this will raise an error in case `snapshot_type` is none of `instance` or `cluster`  (I assume these should be the only accepted values, but we need to enforce this somewhere)